tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -787,6 +787,10 @@ impl Core {
                 cap,
             );
 
+            // Take at least one task since the first task is returned directly
+            // and nto pushed onto the local queue.
+            let n = usize::max(1, n);
+
             let mut synced = worker.handle.shared.synced.lock();
             // safety: passing in the correct `inject::Synced`.
             let mut tasks = unsafe { worker.inject().pop_n(&mut synced.inject, n) };
```

Comment:
In fact, maybe do this and drop the plus one in the first branch of `usize::min`?
```suggestion
            let mut tasks = unsafe { worker.inject().pop_n(&mut synced.inject, n + 1) };
```

Reply:
But that isn't the same. We don't want to always add `1`; we want a min of 1